### ISSUE TITLE
Allow implementations of categories

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#150]
Example:

```
@implementation NSString (MyMethods)

// ...

@end
```

Error: unexpected type `NSString`.

Should be:

```
extension NSString {

// ...

}
```

**Comment by drkameleon** [comment]
Well, I believe it's one of the issues I re-introduced myself in one of the latest releases. Trying to make "intelligent" type recognition, I obviously messed up a few things that *were* working. Let's see...

**Comment by drkameleon** [comment]
(Re)Fixed as of the upcoming 2.3. Well,... hopefully... lol

----

**Example:**
```objective-c
@implementation NSString (MyMethods)

- (void)someMethod {
    // Do something
}

@end
```

**Output:**
```swift

extension String {

	func someMethod() {
		// Do something
	}
	
}
```
